lisa/operating_system.py

```diff
@@ -261,8 +451,59 @@ def _initialize_package_installation(self) -> None:
                 timeout=3600,
             )
 
-    def _install_packages(self, packages: Union[List[str]]) -> None:
-        self._node.execute(f"yum install -y {' '.join(packages)}", sudo=True)
+    def _install_packages(
+        self, packages: Union[List[str]], signed: bool = True
+    ) -> None:
+        command = f"yum install -y {'|'.join(packages)}"
+        if not signed:
+            command.__add__(" --nogpgcheck")
+
+        install_result = self._node.execute(command, sudo=True)
+        # yum returns exit_code=1 if package is already installed.
+        # We do not want to fail if exit_code=1.
+        if install_result.exit_code == 1:
+            self._log.info(
+                f"{' '.join(packages)} is/are already installed."
+                f" stdout: {install_result.stdout}"
+            )
+        elif install_result.exit_code != 0:
+            raise LisaException(
+                f"Failed to install {' '.join(packages)}."
+                f" stdout: {install_result.stdout}"
+            )
+        else:
+            self._log.info(
+                f"{' '.join(packages)} is/are installed successfully."
+                f" stdout: {install_result.stdout}"
+            )
+
+    def _query_package(self, package: str, signed: bool = True) -> bool:
+        command = f"yum list installed | grep -Ei {package}"
+        result = self._node.execute(command, sudo=True)
+        if result.exit_code == 0:
+            return True
+
+        return False
+
+    def _get_os_version(self) -> OsVersion:
+        cmd_result = self._node.execute(
+            cmd="cat /etc/redhat-release", no_error_log=True
+        )
+        if cmd_result.exit_code == 0 and cmd_result.stdout != "":
+            result = cmd_result.stdout
+            for vendor in ["Red Hat", "CentOS", "Fedora", "XenServer"]:
+                if vendor in result:
+                    self._os_version.vendor = vendor
+                    if re.search(r"\brelease\b", result, re.IGNORECASE):
```

Comment:
What it matches for? Add samples here. 
And it looks it can be simpler, for example, use regexp to match a line. Refer to above, there are a lot of regex. But they are not good examples, because they haven't provide real samples.

Reply:
It is just searching for "release", re.search("") searches for a character or substring in a string

Reply:
> It is just searching for "release", re.search("") searches for a character or substring in a string

Made it simpler, searching only for 'release'

Reply:
The next line have some splits too. If they are written like a named capture group pattern, it will be much clear.